captum/optim/_core/loss.py

```diff
@@ -145,25 +338,23 @@ class Diversity(Loss):
 
     def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
         activations = targets_to_values[self.target]
-        return -sum(
-            [
-                sum(
-                    [
-                        (
-                            torch.cosine_similarity(
-                                activations[j].view(1, -1), activations[i].view(1, -1)
-                            )
-                        ).sum()
-                        for i in range(activations.size(0))
-                        if i != j
-                    ]
-                )
-                for j in range(activations.size(0))
-            ]
-        ) / activations.size(0)
+        batch, channels, _, _ = activations.shape

```

Comment:
You can get rid of the two underscores by simply taking the first two indexes of `activations.shape` like so:

```
batch, channels = activations.shape[:2]
```